build.sh

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env bash
+
+# bash error control
+set -o pipefail
+set -e
+
+source bash/inventory.sh
+source bash/common.sh
+source bash/cli.sh
+source bash/docker.sh
+source bash/linuxkit.sh
+source bash/hook-lk-containers.sh
+source bash/shellcheck.sh
+source bash/json-matrix.sh
+source bash/kernel.sh
+source bash/kernel/kernel_default.sh
+source bash/kernel/kernel_armbian.sh
+
+### Initialize the command-line handling. This should behave similar to `make`; PARAM=value pairs are accepted in any order mixed with non-param arguments.
+declare -A -g CLI_PARSED_CMDLINE_PARAMS=()
+declare -a -g CLI_NON_PARAM_ARGS=()
+parse_command_line_arguments "${@}" # which fills the above vars & exports the key=value pairs from cmdline into environment
+# From here on, no more $1 ${1} or similar. We've parsed it all into CLI_PARSED_CMDLINE_PARAMS (already exported in environment) or CLI_NON_PARAM_ARGS
+
+### Configuration
+#declare -g HOOK_KERNEL_OCI_BASE="${HOOK_KERNEL_OCI_BASE:-"ghcr.io/rpardini/tinkerbell/kernel-"}"
+#declare -g HOOK_LK_CONTAINERS_OCI_BASE="${HOOK_LK_CONTAINERS_OCI_BASE:-"ghcr.io/rpardini/tinkerbell/linuxkit-"}"
+declare -g HOOK_KERNEL_OCI_BASE="${HOOK_KERNEL_OCI_BASE:-"quay.io/tinkerbellrpardini/kernel-"}"
+declare -g HOOK_LK_CONTAINERS_OCI_BASE="${HOOK_LK_CONTAINERS_OCI_BASE:-"quay.io/tinkerbellrpardini/linuxkit-"}"
```

Comment:
These need to default to `quay.io/tinkerbell/`. Also if there are new repos that don't exist in `quay.io/tinkerbell` i'll need to create them. Do we maybe want different kernels to be image tag?

Reply:
Yes, those were mostly examples/leftovers from my two setups.
Ref `quay.io` I think you're facing the same dilemma as me: it creates new repos/images with "private" visibility by default?
We could yes twist this so the image is always the same, and only the tag changes across kernel flavors.

Reply:
I changed the default -- but didn't change the naming scheme.